### PR TITLE
implement extraordinary installments step

### PR DESCRIPTION
**Pull request by SelsaErojas**
implement extraordinary installments step

### CODE REVIEW

**Comment by novadeveloper14** [comment]
The title of the PR should contain the task code, in this case cb252

File: package-lock.json
Comment (novadeveloper14) [inline package-lock.json:1]: Is there any reason to update package-lock.json?

File: src/pages/add-prospect/interface.tsx [hunk 1]
```diff
@@ -3,8 +3,10 @@ import { Stack } from "@inubekit/stack";
 import { useMediaQuery } from "@inubekit/hooks";
```
Comment (novadeveloper14) [inline src/pages/add-prospect/interface.tsx:3]: In the project we are organizing the imports in the following way:

External imports (from inubekit, icons, hooks... any library external to the project)
////////space////////
Internal imports with aliases (imports with `@alias` currently the ones we have in the vite.config.ts file)
///////space//////
Internal imports without aliases (such as "./steps/extraordinaryInstallments")

This must be applied to all files that are created or modified. Whenever you see a file that does not comply with this rule, please modify it.

File: src/pages/add-prospect/interface.tsx [hunk 2]
```diff
@@ -76,6 +78,12 @@ export function AddProspectUI(props: AddPositionUIProps) {
           currentStepsNumber.id === stepsAddProspect.generalInformation.id && (
             <RequirementsNotMet />
           )}
+        {currentStepsNumber &&
+          currentStepsNumber.id === stepsAddProspect.specialFertilizers.id && (
```
Comment (novadeveloper14) [inline src/pages/add-prospect/interface.tsx:82]: Let's modify the id specialFertilizers to extraordinaryInstallments

File: src/pages/add-prospect/steps/extraordinaryInstallments/config/types.ts
Comment (novadeveloper14) [inline src/pages/add-prospect/steps/extraordinaryInstallments/config/types.ts:1]: These files where the texts are configured are being left in this path: src\config\pages, which could create a new folder called add-prospect (which represents the assisted)

File: src/pages/add-prospect/steps/extraordinaryInstallments/index.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+import { TableExtraordinaryInstallment } from "@components/data/TableExtraordinaryInstallment";
+
+export interface ExtraordinaryInstallmentsProps {
+  dataTable: IExtraordinaryPayment[];
+  onClickDetails?: (id: string) => void;
+  onClickEdit?: (id: string) => void;
+  onClickEliminate?: (id: string) => void;
+}
+
+export const ExtraordinaryInstallments = (
```
Comment (novadeveloper14) [inline src/pages/add-prospect/steps/extraordinaryInstallments/index.tsx:26]: We are leaving the components as function(), not arrow type functions.

File: src/pages/add-prospect/steps/extraordinaryInstallments/index.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  props: ExtraordinaryInstallmentsProps
+) => {
+  const { dataTable, onClickDetails, onClickEdit, onClickEliminate } = props;
+  const isMobile = useMediaQuery("(max-width:880px)");
+  const [isAddSeriesModalOpen, setAddSeriesModalOpen] = useState(false);
+
+  const openAddSeriesModal = () => {
```
Comment (novadeveloper14) [inline src/pages/add-prospect/steps/extraordinaryInstallments/index.tsx:33]: you could refactor these two functions with a toggle:

```
const toggleAddSeriesModal = () => {
  setAddSeriesModalOpen(!isAddSeriesModalOpen);
};
```

File: src/pages/add-prospect/steps/extraordinaryInstallments/index.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    console.log("Confirmar acción");
+  };
+
+  return (
+    <>
+      <Stack gap="16px" direction="column">
+        <Stack gap="24px" direction="column">
+          <Stack justifyContent="end">
+            <Button
+              type="button"
```
Comment (novadeveloper14) [inline src/pages/add-prospect/steps/extraordinaryInstallments/index.tsx:55]: The button already has some of the parameters that you set by default, determine which ones they are and remove them from the code:

https://github.com/selsa-inube/inubekit-button/blob/main/src/Button/index.tsx

File: src/pages/add-prospect/steps/extraordinaryInstallments/index.tsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+          <Stack justifyContent="center">
+            {dataTable.length > 0 ? (
+              <TableExtraordinaryInstallment
+                data={dataTable}
+                onClickDetails={onClickDetails}
+                onClickEdit={onClickEdit}
+                onClickEliminate={onClickEliminate}
+              />
+            ) : (
+              <Text size="large" type="label" appearance="gray" weight="bold">
```
Comment (novadeveloper14) [inline src/pages/add-prospect/steps/extraordinaryInstallments/index.tsx:81]: The Text already has some of the parameters that you set by default, determine which ones they are and remove them from the code:

https://github.com/selsa-inube/inubekit-text/blob/main/src/Text/index.tsx

Check this in other component texts

File: src/pages/add-prospect/steps/extraordinaryInstallments/index.tsx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+                {TextLabels.NoData}
+              </Text>
+            )}
+          </Stack>
+          <Stack></Stack>
+        </Stack>
+      </Stack>
+      {isAddSeriesModalOpen && (
+        <AddSeriesModal
+          title="Nueva serie"
```
Comment (novadeveloper14) [inline src/pages/add-prospect/steps/extraordinaryInstallments/index.tsx:91]: All Spanish text displayed on the screen should go in a configuration file

**Review by novadeveloper14 — CHANGES_REQUESTED** [review]

File: src/pages/add-prospect/interface.tsx (hunk 2 above)
Comment (SelsaErojas) [reply src/pages/add-prospect/interface.tsx:82]: resolved

File: src/pages/add-prospect/steps/extraordinaryInstallments/config/types.ts
Comment (SelsaErojas) [reply src/pages/add-prospect/steps/extraordinaryInstallments/config/types.ts:1]: resolved

File: src/pages/add-prospect/steps/extraordinaryInstallments/index.tsx (hunk 5 above)
Comment (SelsaErojas) [reply src/pages/add-prospect/steps/extraordinaryInstallments/index.tsx:55]: resolved

File: src/pages/add-prospect/steps/extraordinaryInstallments/index.tsx (hunk 6 above)
Comment (SelsaErojas) [reply src/pages/add-prospect/steps/extraordinaryInstallments/index.tsx:81]: resolved

File: src/pages/add-prospect/steps/extraordinaryInstallments/index.tsx (hunk 7 above)
Comment (SelsaErojas) [reply src/pages/add-prospect/steps/extraordinaryInstallments/index.tsx:91]: resolved

File: src/pages/add-prospect/steps/extraordinaryInstallments/index.tsx (hunk 4 above)
Comment (SelsaErojas) [reply src/pages/add-prospect/steps/extraordinaryInstallments/index.tsx:33]: resolved

File: src/pages/add-prospect/interface.tsx (hunk 1 above)
Comment (SelsaErojas) [reply src/pages/add-prospect/interface.tsx:3]: ok

**Review by novadeveloper14 — APPROVED** [review]

**Review by carin91 — APPROVED** [review]

**Pull request merged**